tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/helm"
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/kubectl"
+	r "github.com/istio-ecosystem/sail-operator/pkg/util/tests/types"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var (
+		resourceAvailable = r.Condition{
+			Type:   "Available",
+			Status: "True",
+		}
+		resourceReconcilied = r.Condition{
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = r.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+
+		crds = []string{
+			// TODO: Find an alternative to this list
+			"authorizationpolicies.security.istio.io",
+			"destinationrules.networking.istio.io",
+			"envoyfilters.networking.istio.io",
+			"gateways.networking.istio.io",
+			"istiorevisions.operator.istio.io",
+			"istios.operator.istio.io",
+			"peerauthentications.security.istio.io",
+			"proxyconfigs.networking.istio.io",
+			"requestauthentications.security.istio.io",
+			"serviceentries.networking.istio.io",
+			"sidecars.networking.istio.io",
+			"telemetries.telemetry.istio.io",
+			"virtualservices.networking.istio.io",
+			"wasmplugins.extensions.istio.io",
+			"workloadentries.networking.istio.io",
+			"workloadgroups.networking.istio.io",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		if skipDeploy {
+			Skip("Skipping the deployment of the operator")
+		}
+
+		When("installed via helm install", func() {
+			BeforeAll(func() {
+				Expect(kubectl.CreateNamespace(namespace)).To(Succeed(), "Namespace failed to be created")
+
+				extraArg := ""
+				if ocp {
+					extraArg = "--set=platform=openshift"
+				}
+
+				Expect(helm.Install("sail-operator", filepath.Join(baseDir, "chart"), "--namespace "+namespace, "--set=image="+image, extraArg)).
+					Should(Succeed(), "Operator failed to be deployed")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetConditions).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable), "Operator deployment is not Available; unexpected Condition")
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=sail-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("sail-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(crds), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
```

Comment:
I'd move this above `It("starts succesfully"` simply because the CRDs are immediately available, whereas a successful start might take some time.